core/templates/dev/head/services/QuestionsListService.ts

```diff
@@ -77,6 +77,10 @@ angular.module('oppia').factory('QuestionsListService', [
           return [];
         }
 
+        if (skillIds[0] === null) {
```

Comment:
Maybe combine this with the previous "if" clause?

Also, this seems a bit loose. I think any entries in skillIds should always be non-null strings. Would that be possible?

Reply:
I could have the default value passed be an empty string. Is that fine?

Reply:
Surely it should just be [] if there is no skill ID being passed? Empty string is also not great because skill IDs are typically not empty strings -- we shouldn't be inventing "fake" sentinel values.

Reply:
Done.